### PR TITLE
engine(externalresource): configure storage only on master

### PR DESCRIPTION
**Pull request by CharlesCheung96**
### What problem does this PR solve?

Issue Number: ref #7304

### What is changed and how it works?
1. configure storage only on master
2. fix some teses in gcrunner
3. add opts for `broker.OpenStorage`

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Comment by CharlesCheung96** [comment]
chaos test verified in :https://github.com/CharlesCheung96/tiflow/actions/runs/3270307081

File: engine/pkg/externalresource/broker/interfaces.go [hunk 1]
```diff
@@ -46,3 +47,17 @@ type Broker interface {
 
 	Close()
 }
+
+type openStorageOptions struct {
+	needEmpty bool
```
Comment (amyangfei) [inline engine/pkg/externalresource/broker/interfaces.go:52]: Does `needEmpty` mean `cleanBeforeOpen`?

File: engine/proto/master.proto [hunk 2]
```diff
@@ -274,3 +275,11 @@ message DeleteJobRequest {
     string tenant_id = 2;
     string project_id = 3;
 }
+
+message QueryStorageConfigRequest {
+}
+
+message QueryStorageConfigResponse {
+    Error err = 1;
+    string address = 2;
```
Comment (amyangfei) [inline engine/proto/master.proto:284]: `address` is not accurate, maybe rename it to `config` is better.

File: engine/pkg/externalresource/broker/interfaces.go (hunk 1 above)
Comment (CharlesCheung96) [reply engine/pkg/externalresource/broker/interfaces.go:52]: Yes, seems `cleanBeforeOpen`  is clearer.

File: engine/executor/dm/worker.go [hunk 3]
```diff
@@ -193,11 +193,18 @@ func (w *dmWorker) CloseImpl(ctx context.Context) {
 // setupStorage opens and configs external storage
 func (w *dmWorker) setupStorage(ctx context.Context) error {
 	rid := dm.NewDMResourceID(w.cfg.Name, w.cfg.SourceID)
-	h, err := w.OpenStorage(ctx, rid)
+	openStorageFunc := func() (broker.Handle, error) {
```
Comment (GMHDBJD) [inline engine/executor/dm/worker.go:196]: how about
```
opts:=make([]opts)
if wt==Dump {
    opts=append(opts, broker.WithEmptyOption())
}
h, err = w.OpenStorage(ctx,rid, opts...)
```

**Review by amyangfei — APPROVED** [review]

**Pull request merged**